### DISCUSSION TITLE
Problem with proper fastapi-users-db-sqlmodel setup / User Creation

### DISCUSSION THREAD

**Discussion by unidesigner** [fastapi-users/fastapi-users#864] [Q&A]
I am trying to change my project to use the new [fastapi-users-db-sqlmodel](https://github.com/fastapi-users/fastapi-users-db-sqlmodel) adapter but I'm running into this exception when trying to create a user:
```
  File ".../fastapi_users/router/register.py", line 62, in register
    created_user = await user_manager.create(user, safe=True, request=request)
  File ".../fastapi_users/manager.py", line 143, in create
    existing_user = await self.user_db.get_by_email(user.email)
  File ".../fastapi_users_db_sqlmodel/__init__.py", line 87, in get_by_email
    return results.first()
AttributeError: 'coroutine' object has no attribute 'first'
```
My setup has probably a problem somewhere, and as it is not yet documented how to use the extension, I tried to derive it from the tests. I am for instance not sure if this is code is correct for `get_user_db`:
```
async_engine = create_async_engine(
    config.Settings().db, future=True)

async_session_maker = sessionmaker(
    async_engine, class_=AsyncSession, expire_on_commit=False)

async def get_async_session() -> AsyncGenerator[AsyncSession, None]:
    async with async_session_maker() as session:
        yield session

async def get_user_db(session: AsyncSession = Depends(get_async_session)):
    yield SQLModelUserDatabase(UserDB, session)
```
and I initialize `FastAPIUsers` with the following user manager function:
```
async def get_user_manager(user_db: SQLModelUserDatabase = Depends(get_user_db)):
    yield UserManager(user_db)
```

Any help is well appreciated.

**Comment by frankie567** [comment]
SQLModel adapter is not officially released as there are some issues regarding UUID (https://github.com/tiangolo/sqlmodel/issues/25).

Notice that there are two classes in this adapter: `SQLModelUserDatabase`, the sync version, and `SQLModelUserDatabaseAsync`, the async one. So this should work better:

```py
async def get_user_db(session: AsyncSession = Depends(get_async_session)):
    yield SQLModelUserDatabaseAsync(UserDB, session)
```

**Reply by unidesigner** [reply]
Thank you! This solved the issue. It seems like a lot of PRs in the queue for sqlmodel. Hope they will get merged sometime soon. Really like this library and would like to use it in production.
